### DISCUSSION TITLE
Crea Temas de Blogger con Hamlet

### DISCUSSION THREAD

**Discussion by giscus** [zkreations/data#20] [Announcements]
# Crea Temas de Blogger con Hamlet

Entorno de desarrollo para desarrollar plantillas de Blogger utilizando Handlebars.

https://www.zkreations.com/2023/04/crea-temas-de-blogger-con-hamlet.html

**Comment by danieIabel** [comment]
He archivado la entrada de Canvas ya que considero que dejo de ser util. Hamlet será su reemplazo.

**Comment by ilimon** [comment]
Any video tutorial on how to install?

**Reply by danieIabel** [reply]
First, you need to install [Node.js (LTS)](https://nodejs.org/es/) and [Git](https://git-scm.com/downloads) on your computer. Then, clone the GitHub repository and run the command "npm install" to install the dependencies.

**Comment by TakoMics** [comment]
Excelente tutorial Daniel. 

Tal como dije hace unas entradas pasadas, creo que aún (espero que algún día eso cambie, claro está), no estoy en condiciones para aventurarme a crear un tema para Blogger.

Aun así, empezar a estudiar la sintaxis, la estructura y el desarrollo de variables y el código en sí, considero que es lo ideal y a su vez muy útil para trabajar con fluidez en el set de plantillas ya existentes.

Por cierto, la plantilla de demostración está genial. Me parece una buena sucesora de las plantillas Simple.xml y Basico.xml.


Es posible que se me haya pasado leyendo la documentación actual de las plantillas en la tienda, pero de no estar en cómo crear algo similar, sería interesante que en un próximo tutorial nos enseñaras cómo hacer algo así en medio de una entrada o una página estática:

(Adjunto una captura de referencia en otra web —siéntete en la libertad de eliminar el comentario si infringe alguna cláusula del chat—) https://i.imgur.com/mDRVm2h.png

Muchas gracias Daniel.

**Reply by danieIabel** [reply]
Que tal amigo, tengo una ruta que estoy siguiendo para finalmente llegar a un tutorial. por ahora estoy compartiendo proyectos que facilitan la creación de plantillas y plugins para Blogger, en conjunto con algunas entradas donde comparto "funciones" creadas con código de Blogger, también explicadas en el mismo artículo.

Cuando termine con todo eso, tendre contenido académico suficiente que puedo referenciar en lo que finalmente será el "Tutorial" para crear una plantilla. Confieso que me tomo mi tiempo en este tipo de cosas, ya que mi prioridad siempre serán las personas que deciden trabajar conmigo, aun asi, si veo mucho apoyo en este tipo de contenido, evidentemente le dedicare mas tiempo. Un saludo!.

**Reply by TakoMics** [reply]
Entonces excelente. He entendido en cabalidad a lo que quieres llegar, para así crear una especie de “biblioteca” a la que puedas hacer referencia al momento de armar el tutorial de creación de plantillas en sí.

Pues, todo a tu ritmo y como se te haga más cómodo. Acá nosotros estamos disfrutando del contenido, aprendiendo y tomando nota, ya que la información es valiosísima.

Saludos.

**Reply by danieIabel** [reply]
Gracias amigo, según mis planes espero terminar todo en el lapso de dos mes. Lo bueno de los artículos es que a diferencia de los videos, un artículo si se puede mantener actualizado.

**Comment by danieIabel** [comment]
He actualizado el entorno con mejoras y nuevos añadidos que espero ayuden a mejorar aún mas el desarrollo de un tema de Blogger. La información de lo nuevo fue actualizada en el post.

**Comment by ghost** [comment]
¡Genial amigo!

**Comment by danieIabel** [comment]
He actualizado el entorno agregando una nueva característica que ayuda a separar la lógica del código de Blogger (b:skin) de los estilos (sass). Recomiendo revisar del árbol de contenido "Datos adicionales".

**Comment by danieIabel** [comment]
He actualizado de nuevo el entorno, esta vez removiendo la necesidad de usar variables de entorno, ya que solo se usaba para decidir el modo de compilado, en su lugar he agregado dos comandos:

```bash
npm run dev
npm run prod
```

De esa forma puedes decidir en qué modo empezar a compilar. Sí se usa el comando `npm run compile`, el resultado siempre será producción.

También se agregaron algunos helpers que pueden ser de utilidad para realizar comprobaciones y crear mejores condiciones.

**Comment by danieIabel** [comment]
He actualizado el repositorio. Ahora el entorno que compila la plantilla de Blogger es un paquete de node independiente, lo que hace que sea mas fácil de mantener actualizado. Todo lo que ya tenias con al version anterior funciona en este paquete, pero además podrás usar todas las características del paquete que puedes leer en esta entrada o en Github (Ingles): [github.com/zkreations/hamlet-builder](https://github.com/zkreations/hamlet-builder). Aún así si experimentas algún problema no dudes en comentarlo.
